lisa/notifiers/text_result.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import Any, List, Type, Union
+
+from lisa import notifier, schema
+from lisa.runner import print_results
+from lisa.testsuite import TestResult, TestResultMessage
+
+
+class TextResult(notifier.Notifier):
+    """
+    Creating log notifier to dump text formatted results for easier
+    view in editing mode. The original log is complete but too long to
+    check only the summary.
+    """
+
+    @classmethod
+    def type_name(cls) -> str:
+        return "text_result"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return schema.Notifier
+
+    def _received_message(self, message: TestResultMessage) -> None:
+        if message.is_completed:
+            self.received_messages.append(Union(None, message))
+
+    def _subscribed_message_type(self) -> List[Type[notifier.MessageBase]]:
+        return [TestResultMessage]
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self.result_path = """
+        {constants.RUN_LOCAL_PATH}/lisa-{constants.RUN_ID}-result.txt
+        """
+        result_file = open(self.result_path, "x")
```

Comment:
What's the reason to have open/close here? Because the logic is changed, if it needs to check if the file exists here, just check it.